lib/datadog/appsec/contrib/rack/request_middleware.rb

```diff
@@ -33,6 +33,21 @@ def call(env)
             ready = false
             context = nil
 
+            # If a customer has mounted a nested Rack app within their main app
+            # and instrumented boths apps:
+            #
+            # Datadog.configure do |c|
+            #   c.appsec.enabled = true
+            #   c.appsec.instrument :rails
+            #   c.appsec.instrument :sinatra
+            # end
+            #
+            # We have to avoid calling processor.activate_context twice.
+            # To achive that we check the value of env['datadog.waf.context']
+            # If the waf context is part of the env, it means that we are on the nested app
+            # context. In that case we need process the request without any new appsec related code.
```

Comment:
I feel it can be distilled into:

```suggestion
            # For a given request, keep using the first Rack stack context for
            # nested apps. Don't set `context` local variable so that on popping
            # out of this nested stack we don't finalize the parent's context
```